history/src/block_stores/postgres_block_store.rs

```diff
@@ -1 +1,106 @@
+use std::sync::Arc;
 
+use async_trait::async_trait;
+use itertools::Itertools;
+use solana_lite_rpc_core::{
+    structures::{epoch::EpochCache, produced_block::ProducedBlock},
+    traits::block_storage_interface::BlockStorageInterface,
+};
+use solana_rpc_client_api::config::RpcBlockConfig;
+use solana_sdk::{slot_history::Slot, stake_history::Epoch};
+use tokio::sync::RwLock;
+
+use crate::postgres::{
+    postgres_block::PostgresBlock, postgres_session::PostgresSessionCache,
+    postgres_transaction::PostgresTransaction,
+};
+
+#[derive(Default, Clone, Copy)]
+pub struct PostgresData {
+    from_slot: Slot,
+    to_slot: Slot,
+    current_epoch: Epoch,
+}
+
+pub struct PostgresBlockStore {
+    session_cache: PostgresSessionCache,
+    epoch_cache: EpochCache,
+    postgres_data: Arc<RwLock<PostgresData>>,
+}
+
+impl PostgresBlockStore {
+    pub async fn start_new_epoch(&self, schema: &String) {
+        // create schema for new epoch
+        let session = self
+            .session_cache
+            .get_session()
+            .await
+            .expect("should get new postgres session");
+
+        let statement = format!("CREATE SCHEMA {};", schema);
+        if let Err(e) = session.execute(&statement, &[]).await {
+            log::error!("Error creating a schema for {} error {}", schema, e);
```

Comment:
I think the errors should be return because is the db is not created there's no need to store block and other actions must be done ASAP.